packages/sanity/src/core/form/inputs/files/common/FileInputButton/styles.ts

```diff
@@ -15,7 +16,7 @@ export const FileButton = styled(Button).attrs({forwardedAs: 'label'})(
         }
       }
 
-      & input {
+      & > span:nth-child(2) {
```

Comment:
the button children introduces a span, so we need to hide that and not the actual input.